compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -10529,6 +10527,19 @@ OMR::Z::TreeEvaluator::arraycmpHelper(TR::Node *node,
          cg->decReferenceCount(source2Node);
          }
 
+      if (baseSource1Ref->getOffset() == baseSource2Ref->getOffset())
```

Comment:
Hmm, I do see what you are trying to do here. Wouldn't this miss, a case where we have different offset and base register values, but both memory reference would point to same location ?

Reply:
I can load address to register to cover that case but I am not sure if it worth it since the probability of that scenario doesn't seem to be high. I am still trying to figure out how the created memRefs could have non zero offset. We create a MemRef from node which I think it just evaluate the node and set the base register (the provided displacement is zero)

Reply:
I am talking about the case where we do have the symbol reference on the node. In that case, you are comparing the offset and if the offset at compile time is same, u generate the runtime test to check if the value in base register is same or not. 
For example we have a case where offset1 + base1 is equal to offset2 + base2 but offset1 != offset2 and same with base register. 

If you meant to have a quick optimize test specific to case where at compile time you know offset are same and at run time you check in that case if the base values are same or not then this branch is justified, but this will simply covering a single case.

We should look into the series of instruction generated here to know if we can inspect the memory reference pointing to the same location or not. 

Can you share the generated instructions here?

Reply:
I am aware of this way of comparing, I do want to take a look at the instruction selection log to see if there is a better way.